### ISSUE TITLE
Selector does not work when the props parameter is undefined

### ISSUE THREAD

**Issue by Odonno** [ngrx/platform#2024]
I created a selector with props and I used it but it did not work as expected. When I passed an "existing" value to the props param, it works like a charm but when I passed `undefined`, the selector did not work at all.

## Minimal reproduction of the bug/regression with instructions:

```ts
const selectFilteredValues = createSelector(
    selectValues,
    (values, filter) => {
        if (filter) { return values.filter(...); }
        return ["DEFAULT_VALUE"];
    }
);

const filter = undefined;
const filteredValues$ = this.store.pipe(
    select(selectFilteredValues, filter)
);
```

When `filter` is a valid string, it works. When `filter` is `undefined`, it does not even pass in the selector...

## Versions of NgRx, Angular, Node, affected browser(s) and operating system(s):

```
"@angular/animations": "~8.1.1",
"@angular/common": "~8.1.1",
"@angular/compiler": "~8.1.1",
"@angular/core": "~8.1.1",
"@angular/forms": "^8.1.1",
"@angular/platform-browser": "~8.1.1",
"@angular/platform-browser-dynamic": "~8.1.1",
"@angular/pwa": "~0.801.1",
"@angular/service-worker": "~8.1.1",
"@ngrx/effects": "^8.1.0",
"@ngrx/store": "^8.1.0",
"@ngrx/store-devtools": "^8.1.0",
"core-js": "^2.5.4",
"jquery": "^3.3.1",
"ngrx-signalr": "^8.0.14",
"ngx-take-until-destroy": "^5.4.0",
"rxjs": "~6.5.2",
"signalr": "^2.4.0",
"tslib": "^1.9.0",
"zone.js": "~0.9.1"
```

## I would be willing to submit a PR to fix this issue

[x] Yes (Assistance is provided if you need help submitting a pull request)
[ ] No

**Comment by timdeschryver** [comment]
Can you define "did not work at all" please and provide a reproduction?
From what I can see, it behaves correctly.

**Comment by Odonno** [comment]
Well, I assumed we could pass `any` object to the selector props but `undefined` is checked to see if we are using a `Selector` or a `SelectorWithProps`..

https://github.com/ngrx/platform/blob/136c5eb1e7ba45adc7d8c105e21a45fed7546ab6/modules/store/src/selector.ts#L511

@timdeschryver So, per my previous example, when you apply the `selectFilteredValues` selector with an `undefined` value, it creates a `Selector` without props. But I do not know why it does not trigger the selector at all...

So, I suppose it is a way better pratice to use an object that contains properties (which then can be `undefined`). From that knowledge, we could switch from a `any` type definition to `object | undefined` type definition for the props, what do you think?

**Comment by timdeschryver** [comment]
I'm still not sure, if `props` undefined it does execute the selector with `undefined` `props` because there are no `props`.
From what I can see [it does work as intended](https://stackblitz.com/edit/ngrx-seed-urpzqc?file=src/app/app.module.ts).

**Comment by Odonno** [comment]
Well, I made a fresh new test project and tried with an `undefined` filter. It does work as expected. 

But on my current work project, it works if I set a valid string as props but it does not pass into the selector at all when the props are `undefined` but again if I create an object like `{ filter: undefined }`, it works... 

Maybe it is due to the migration from ng7 to ng8 or something else. I need to figure that out.
